webpack.config.js

```diff
@@ -1,74 +1,113 @@
-// const ForkTsCheckerWebpackPlugin = require("fork-ts-checker-webpack-plugin");
+const isProd = process.env.NODE_ENV === "production";
+const ASSET_PATH = isProd ? "" : "/";
+const version = packageJson.version;
 
-const {
-  dependencySrcMapRules,
-  stylingRules,
-  svgUrlRules,
-  getContext,
-  getOptimization,
-  getResolve,
-  tsRules
-} = require("./webpack.rules");
+// Calculate hash based on content, will be used when generating production 
+// bundles
+const cssLoader = {
+  loader: "css-loader",
+  options: {
+    modules: {
+      auto: true,
+      localIdentName: isProd ? "[hash:base64]" : "[name]__[local]--[hash:base64:5]",
+    },
+  },
+};
 
-const isProd = process.env.NODE_ENV === "production";
+const rules = [
+  {
+    test: /\.js$/,
+    use: "source-map-loader",
+    enforce: "pre",
+    exclude: /node_modules/,
+  },
+  {
+    test: /\.(css|less)$/,
+    use: [
+      isProd ? MiniCssExtractPlugin.loader : "style-loader",
+      cssLoader,
+      "less-loader",
+    ],
+  },
+  {
+    test: /\.svg(\?v=\d+\.\d+\.\d+)?$/,
+    type: "asset",
+    parser: {
+      dataUrlCondition: {
+        maxSize: 10 * 1024, // 10kb
+      },
+    },
+  },
+  {
+    test: /\.tsx?$/,
+    exclude: /node_modules/,
+    use: [
+      {
+        loader: "ts-loader",
+        options: {
+          transpileOnly: true,
+          experimentalWatchApi: true,
+        },
+      },
+    ],
+  },
+];
 
-const basicConfig = {
+module.exports = {
+  mode: isProd ? "production" : "development",
+  // generate a sourcemap for production, in dev we generate mapping faster
+  devtool: isProd ? "source-map" : "eval-cheap-module-source-map",
   devServer: {
     port: 8000,
+    // enable hot module replacement
+    hot: true,
   },
-  devtool: isProd ? false : "source-map",
   entry: ["src/index.tsx", "src/main.tsx"],
-  watch: false,
-  ...getContext(__dirname),
-
   output: {
     path: path.resolve(__dirname, "dist"),
-    filename: "[name].js",
-    publicPath: "/"
+    filename: isProd ? "[name].js" : "[name].js",
+    publicPath: ASSET_PATH,
+    clean: true,
   },
-
-  module: {
-    rules: [
-      ...dependencySrcMapRules,
-      ...stylingRules,
-      ...svgUrlRules,
-      ...tsRules,
-    ]
-  },
-
+  module: { rules },
   resolve: {
-    ...getResolve(__dirname)
+    modules: ["node_modules", path.resolve(__dirname)],
+    extensions: [".tsx", ".ts", ".jsx", ".js", ".less", ".css"],
   },
-
   plugins: [
-    new HtmlWebpackPlugin({
-      title: "conda-store"
+    new HtmlWebpackPlugin({ title: "conda-store" }),
+    new MiniCssExtractPlugin({
+      filename: "[name].css",
     }),
-    new MiniCssExtractPlugin(),
     new Dotenv(),
-    new webpack.EnvironmentPlugin(['REACT_APP_VERSION'])
+    new webpack.EnvironmentPlugin(["REACT_APP_VERSION"]),
+    new webpack.ids.HashedModuleIdsPlugin(),
+    new webpack.DefinePlugin({
+      'process.env.VERSION': JSON.stringify(version),
+    }),
+    new webpack.BannerPlugin({
```

Comment:
What does this do? Could you add a quick comment? I'm guessing it adds a comment at the top of the file? But in both prod and dev?